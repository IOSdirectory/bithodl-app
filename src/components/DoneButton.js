import React, { Component } from 'react';
import { StyleSheet, TouchableOpacity, Text } from 'react-native';
import PropTypes from 'prop-types';
import StyledText from './StyledText';

const styles = StyleSheet.create({
  button: {
    padding: 10
  },
  text: {
    fontSize: 16,
    fontWeight: '500',
    color: '#D49146'
  }
});

export default class DoneButton extends Component {
  render() {
    return (
      <TouchableOpacity onPress={this.props.onPress} style={styles.button}>
        <Text>
          <StyledText style={styles.text}>
            Done
          </StyledText>
        </Text>
      </TouchableOpacity>
    );
  }
}

DoneButton.propTypes = {
  onPress: PropTypes.func
};
