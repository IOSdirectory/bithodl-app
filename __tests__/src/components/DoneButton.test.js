import React from 'react';
import renderer from 'react-test-renderer';
import DoneButton from '../../../src/components/DoneButton';

describe('DoneButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <DoneButton />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
