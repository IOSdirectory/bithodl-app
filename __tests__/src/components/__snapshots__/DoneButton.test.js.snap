// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoneButton renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Object {
            "color": "#D49146",
            "fontSize": 16,
            "fontWeight": "500",
          },
        ]
      }
    >
      Done
    </Text>
  </Text>
</View>
`;
